src/panic.rs

```diff
@@ -1,31 +1,41 @@
-use core::mem;
-
-#[repr(C)]
-struct PanicPayload {
-	file_ptr: *const u8,
-	file_len: usize,
-	line: u32,
-	col: u32,
-}
+#![cfg(not(feature = "std"))]
+
+use {Vec};
+use byteorder::{LittleEndian, ByteOrder};
 
 #[lang = "panic_fmt"]
 pub fn panic_fmt(_fmt: ::core::fmt::Arguments, file: &'static str, line: u32, col: u32) -> ! {
 	extern "C" {
 		fn panic(payload_ptr: *const u8, payload_len: u32) -> !;
 	}
+
+	#[cfg(feature = "panic_with_msg")]
+	let message = format!("{}", _fmt);
+
+	#[cfg(not(feature = "panic_with_msg"))]
+	let message = ::alloc::String::new();
+
+	let mut payload = Vec::new();
```

Comment:
could be Vec::with_capacity(message.as_bytes().len() + file.as_bytes().len() + 8)